react/ProductList.tsx

```diff
@@ -55,49 +66,68 @@ const ItemContextWrapper = memo<ItemWrapperProps>(function ItemContextWrapper({
       onSetManualPrice: (price: number, itemIndex: number) =>
         onSetManualPrice(price, itemIndex),
     }),
-    [item, loading, onQuantityChange, onRemove]
+    [
+      item,
+      itemIndex,
+      loading,
+      onQuantityChange,
+      onRemove,
+      onSetManualPrice,
+      shouldAllowManualPrice,
+    ]
   )
 
   return <ItemContextProvider value={context}>{children}</ItemContextProvider>
 })
 
-const ProductList: StorefrontFunctionComponent<Props> = ({
-  allowManualPrice,
-  items,
-  loading,
-  userType,
-  onQuantityChange,
-  onRemove,
-  onSetManualPrice,
-  children,
-}) => {
-  const handles = useCssHandles(CSS_HANDLES)
+const ProductGroup: StorefrontFunctionComponent<Props> = props => {
+  const { items, renderOnView, userType, allowManualPrice, children } = props
   const shouldAllowManualPrice =
     allowManualPrice && userType === CALL_CENTER_OPERATOR
 
-  const [availableItems, unavailableItems] = items.reduce<Item[][]>(
-    (acc, item) => {
-      acc[item.availability === AVAILABLE ? 0 : 1].push(item)
-      return acc
-    },
-    [[], []]
+  const { hasBeenViewed, dummyElement } = useRenderOnView({
+    lazyRender: true,
+    offset: 900,
+  })
+
+  if (renderOnView && (!hasBeenViewed || items.length === 0)) {
+    return dummyElement
+  }
+
+  return (
+    <>
+      {items.map((item: ItemWithIndex) => (
+        <ItemContextWrapper
+          key={item.uniqueId + item.sellingPrice}
+          item={item}
+          itemIndex={item.index}
```

Comment:
Since now we can access the `index` of a given item by doing `item.index`, I think we can remove this `itemIndex` prop from the item's context. But we can do this later in another PR.